### PR TITLE
Update Queries.pm

### PR DESCRIPTION
**Pull request by AF-1**
add lossless query option

### CODE REVIEW

File: Slim/Control/Queries.pm [hunk 1]
```diff
@@ -4822,6 +4822,7 @@ my %colMap = (
 	y => 'tracks.year',
 	m => 'tracks.bpm',
 	M => sub { $_[0]->{'tracks.musicmagic_mixable'} ? 1 : 0 },
+ 	Q => 'tracks.lossless',
```
Comment (mherger) [inline Slim/Control/Queries.pm]: Please make this a boolean like in the above `M` case to make sure we get either `1` or `0`, but no undefined or other value.

File: Slim/Control/Queries.pm [hunk 2]
```diff
@@ -4771,7 +4771,7 @@ my %tagMap = (
 	  'M' => ['musicmagic_mixable', '',            'musicmagic_mixable'], #musicmagic_mixable
 	                                                                    #musicbrainz_id
 	                                                                    #lastplayed
-	                                                                    #lossless
+	  'Q' => ['lossless',         'LOSSLESS',      'lossless'],         #lossless
 	  'w' => ['lyrics',           'LYRICS',        'lyrics'],           #lyrics
 	  'R' => ['rating',           'RATING',        'rating'],           #rating
 	  'O' => ['playcount',        'PLAYCOUNT',     'playcount'],        #playcOunt
```
Comment (mherger) [inline Slim/Control/Queries.pm]: Oh, while you're at it: could you please fix the typo in the comment? 😂

**Comment by AF-1** [comment]
BTW is x (remote) not working? Querying x for a local track returns nothing here. Should return "0", no? Can you reproduce this?

**Review by michaelherger — COMMENTED** [review]
I'm sorry... one more request: could you please document this around https://github.com/Logitech/slimserver/blob/5f2a72212cb867c61fa1c3508bbc19dc6a525dfd/HTML/EN/html/docs/cli-api.html#L5318? Thanks in advance!

**Comment by michaelherger** [comment]
> BTW is x (remote) not working? Querying x for a local track returns nothing here. Should return "0", no? Can you reproduce this?

I'm getting a `0` for a local track:

```
status 0 1 tags:x
00:04:... status 0 1 tags:x player_name:SqueezePlay-M1 player_connected:1 player_ip:192.168.0.5:62905 power:1 signalstrength:0 mode:play time:6.0673002166748 rate:1 duration:229.72 can_seek:1 replay_gain:0 mixer volume:63 playlist repeat:0 playlist shuffle:0 playlist mode:off seq_no:2 playlist_cur_index:1 playlist_timestamp:1695132520.03351 playlist_tracks:16 digital_volume_control:1 playlist index:0 id:319618 title:Element of Freedom (intro) remote:0
```

**Comment by AF-1** [comment]
Ok, I used `songinfo`, not `status`. Not getting anything for remote. Hmm...

**Comment by AF-1** [comment]
Is there any special reason why the remote tag value is set to **undef** if it's 0? (at least in songinfoquery / _songData)

https://github.com/Logitech/slimserver/blob/public/8.4/Slim/Control/Queries.pm#L5098

If you remove the x part, `songinfo` returns a 0 for local tracks.

**Comment by mherger** [comment]
That's odd indeed. But that change is from the really early days of the CLI (16 years ago!), done by a contractor at the time. I would be surprised if it hurt if that was reverted. Somewhere down the line something would remove undefined values. So instead of `remote:0` we now have no `remote` value.

**Pull request merged**